contracts/Stonks.sol

```diff
@@ -79,44 +83,49 @@ contract Stonks is IStonks, AssetRecoverer {
         orderParameters = OrderParameters({
             tokenFrom: tokenFrom_,
             tokenTo: tokenTo_,
-            orderDurationInSeconds: uint32(orderDurationInSeconds_),
-            marginInBasisPoints: uint16(marginInBasisPoints_),
-            priceToleranceInBasisPoints: uint16(priceToleranceInBasisPoints_)
+            orderDurationInSeconds: orderDurationInSeconds_.toUint32(),
+            marginInBasisPoints: marginInBasisPoints_.toUint16(),
+            priceToleranceInBasisPoints: priceToleranceInBasisPoints_.toUint16()
         });
     }
 
     /**
      * @notice Initiates a new trading order by creating an Order contract clone with the current token balance.
      * @dev Transfers the tokenFrom balance to the new Order instance and initializes it with the Stonks' manager settings for execution.
      */
-    function placeOrder() external onlyAgentOrManager returns (address) {
+    function placeOrder(uint256 minBuyAmount_) external onlyAgentOrManager returns (address) {
+        if (minBuyAmount_ == 0) revert InvalidAmount();
+
         uint256 balance = IERC20(orderParameters.tokenFrom).balanceOf(address(this));
 
         // Contract needs to hold at least 10 wei to cover steth shares issue
         if (balance <= MIN_POSSIBLE_BALANCE) revert MinimumPossibleBalanceNotMet();
 
         Order orderCopy = Order(Clones.clone(orderSample));
         IERC20(orderParameters.tokenFrom).safeTransfer(address(orderCopy), balance);
-        orderCopy.initialize(manager);
+        orderCopy.initialize(minBuyAmount_, manager);
+
+        emit OrderContractCreated(address(orderCopy));
```

Comment:
Unfortunately we can't support event interface for `cow-watch-tower` because it requires to pack specific metadata and inherit interfaces we don't use. Moreover, there is no guarantee that it will be working in a stable way and for a long time. The solution suggested by @avsetsin has been abandoned and replaced with `cow-watch-tower`. 
As a result we decided to integrate offchain order creation to EasyTrack UI and support it by ourselves.